finagle-benchmark/src/main/scala/com/twitter/finagle/loadbalancer/RoundRobin.scala

```diff
@@ -0,0 +1,84 @@
+package com.twitter.finagle.loadbalancer
+
+import com.twitter.finagle.{ClientConnection, NoBrokersAvailableException, Service, ServiceFactory}
+import com.twitter.finagle.service.FailingFactory
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.finagle.util.OnReady
+import com.twitter.util.{Activity, Future, Promise, Time}
+
+object RoundRobinBalancer {
```

Comment:
The only benefit is having fewer implementations to maintain. I'd just prefer waiting to promote it into core until we are sure that it solves the use cases you have in mind.